vcd/resource_vcd_vapp_vm.go

```diff
@@ -616,8 +707,22 @@ func resourceVcdVAppVmRead(d *schema.ResourceData, meta interface{}) error {
 	}
 
 	d.Set("name", vm.VM.Name)
-	if len(vm.VM.NetworkConnectionSection.NetworkConnection) > 0 {
+	networks := d.Get("networks").([]interface{})
+	network := d.Get("network_name").(string)
+	switch {
+	// network_name is not set. networks is set in config
+	case network != "" && len(networks) == 0:
 		d.Set("ip", vm.VM.NetworkConnectionSection.NetworkConnection[0].IPAddress)
+		d.Set("mac", vm.VM.NetworkConnectionSection.NetworkConnection[0].MACAddress)
+	case network == "" && len(networks) > 0:
+		var networks []map[string]interface{}
+		for index, net := range d.Get("networks").([]interface{}) {
+			n := net.(map[string]interface{})
+			n["ip"] = vm.VM.NetworkConnectionSection.NetworkConnection[index].IPAddress
+			n["mac"] = vm.VM.NetworkConnectionSection.NetworkConnection[index].MACAddress
+			networks = append(networks, n)
+			d.Set("networks", networks)
```

Comment:
Are you sure you need to set it inside the loop?
If I understand correctly this will trigger statefile updates everytime.

Reply:
Fixed.